### PR TITLE
Fix watchers tab

### PR DESCRIPTION
**Pull request by NobodysNightmare**
## OpenProject work package

https://community.openproject.org/work_packages/19605
## Description

The issue that username and avatar are not displayed was solved quite easily by reintroducing a deleted directive.

However, I still think that the watchers tab is far from ideal:

watchers-tab

The user name is displayed in three different places:
- where you can delete the user
- above the select2
- inside the select2

IMHO only the first place makes sense... or only the last one, but then it is questionable why we need a whole tab at all, because the tab would basically only contain a single select2.

What's also weird is that the text above the select2 and the select2 itself are not updated instantly, but need a reload of the watchers tab to occur.

I think there are no visuals for that tab, so nothing is defined and everything is allowed. Not sure if we got time to improve stuff, though.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://nobodysnightmare-feature-fix-watchers-tab.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://nobodysnightmare-feature-fix-watchers-tab.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by NobodysNightmare** [comment]
@0xF013 As you are the one that added it:
Do you know the purpose of the div right above the select2, that lists the user names?

**Comment by 0xF013** [comment]
@NobodysNightmare I guess I reintroduced that line after a merge, it was supposed to be deleted

**Comment by NobodysNightmare** [comment]
@0xF013 Then that is what I will do.

My first thought was a relation to accessibility, but at least my chromevox did make no use of it.

**Comment by 0xF013** [comment]
@NobodysNightmare oh sure, sure, I was thinking of accessiblity

**Comment by NobodysNightmare** [comment]
@0xF013 In fact it is used by the select2: `aria-describedby="watchers_selected_list"`...

Is it possible that I should just add a class to it? (e.g. `hidden-for-sighted`?)

**Comment by 0xF013** [comment]
@NobodysNightmare wtf, maybe some CSS rule got removed. sure, hidden-for-sighted should be there. I am out of ideas to cover my ass right now

**Comment by 0xF013** [comment]
@NobodysNightmare also you might wanna change aria-describedby to aria-labelledby

**Comment by myabc** [comment]
@NobodysNightmare it's not possible to delete watchers within the select2 box.

**Comment by myabc** [comment]
The UI can be simplified dramatically (still needs a few tweaks):
seeded_project_-_work_packages_-_openproject

**Comment by NobodysNightmare** [comment]
@myabc This is what I meant:

> or only the last one, but then it is questionable why we need a whole tab at all, because the tab would basically only contain a single select2.

I don't think that this would make for an aesthetic experience.

That is why I would prefer to have the list with users and their delete buttons (the thing that I repair with this PR) and below that list only a single select2 box, used to add users to the aforementioned list. That means the select2 will not mirror the whole list of selected users.

However, that is only my personal cup of tea, your idea is perfectly valid too. And for the sake of this bug we need to do neither...

**Comment by NobodysNightmare** [comment]
@myabc 

> @NobodysNightmare it's not possible to delete watchers within the select2 box.

I think you are talking about this issue: https://github.com/opf/openproject/pull/2780

**Comment by myabc** [comment]
> And for the sake of this bug we need to do neither...

OK, fair enough. We should add a separate WP/PR to clean up this view.

File: frontend/app/ui_components/index.js [hunk 1]
```diff
@@ -102,6 +102,7 @@ angular.module('openproject.uiComponents')
     './toggled-multiselect-directive')])
   .directive('toolbar', require('./toolbar-directive'))
   .constant('ESC_KEY', 27)
+  .directive('userField', [require('./user-field-directive')])
```
Comment (myabc) [inline frontend/app/ui_components/index.js:105]: @NobodysNightmare can we rename this to `userAvatar`? it would better reflect what the directive does.

Comment (myabc) [inline frontend/app/ui_components/index.js:105]: @NobodysNightmare nvm.. will merge as is. we can shift the rename to a separate PR.

**Pull request merged**